fdbserver/DataDistribution.actor.cpp

```diff
@@ -1538,11 +1575,78 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 		return true;
 	}
 
+	int calculateHealthyServerCount() {
+		int serverCount = 0;
+		for (auto i = server_info.begin(); i != server_info.end(); ++i) {
+			if (!server_status.get(i->first).isUnhealthy()) {
+				++serverCount;
+			}
+		}
+		return serverCount;
+	}
+
+	int calculateHealthyMachineCount() {
+		int totalHealthyMachineCount = 0;
+		for (auto& m : machine_info) {
+			if (isMachineHealthy(m.second)) {
+				++totalHealthyMachineCount;
+			}
+		}
+
+		return totalHealthyMachineCount;
+	}
+
+	// Sanity check
+	bool isServerTeamNumberCorrect(Reference<TCMachineTeamInfo>& mt) {
+		int num = 0;
+		bool ret = true;
+		for (auto& team : teams) {
+			if (team->machineTeam->machineIDs == mt->machineIDs) {
+				++num;
+			}
+		}
+		if (ret == false) {
+			TraceEvent(SevError, "ServerTeamNumberOnMachineIncorrect")
+			    .detail("MachineTeam", mt->getMachineIDsStr())
+			    .detail("ServerTeamsSize", mt->serverTeams.size())
+			    .detail("CountedServerTeamNumber", num);
+		}
+		return ret;
+	}
+
+	// Find the machine team with the least number of server teams
+	int getMachineTeamWithLeastProcessTeams(Reference<TCMachineTeamInfo>& ret) {
+		int minNumProcessTeams = std::numeric_limits<int>::max();
+
+		for (auto& mt : machineTeams) {
+			ASSERT(isServerTeamNumberCorrect(mt));
```

Comment:
this is too expensive to do all the time, add the EXPENSIVE_VALIDATION check to this assert